### DISCUSSION TITLE
about type system

### DISCUSSION THREAD

**Discussion by rraammiinn** [imba/imba#821] [Q&A]
does it supports static typing ?

**Comment by haikyuu** [comment]
A limited subset of typescript is supported, mainly typing variables, class properties and method parameters using 
```imba
const a\number = 23
def add n\number, m\string
    n + m
```

**Reply by rraammiinn** [reply]
why not : instead of \ ?

**Reply by haikyuu** [reply]
because `a: 1` compiles to `{a:1}`
